test/GPv2Encoding.test.ts

```diff
@@ -327,6 +327,169 @@ describe("GPv2Encoding", () => {
+      // easier to spot. 0x11 is used to avoid having bytes with zeroes in their
+      // hex representation.
+      const interaction = {
+        target: ethers.utils.getAddress(fillDistinctBytes(20, 0x11)),
+        callData: fillDistinctBytes(42, 0x11 + 20),
+      };
+
+      const encoder = new SettlementEncoder(testDomain);
+      await encoder.encodeInteraction(interaction);
+
+      const numInteractions = 1;
+      const [decodedInteractions] = await encoding.decodeInteractionsTest(
+        encoder.encodedInteractions,
+        numInteractions,
+      );
+
+      expect(decodedInteractions.length).to.equal(1);
+      expect(decodedInteractions[0].target).to.equal(interaction.target);
+      expect(decodedInteractions[0].callData).to.equal(interaction.callData);
+    });
+
+    it("should round-trip encode multiple interactions", async () => {
+      // Note: all fields should use distinct bytes as much as possible to make
+      // decoding errors easier to spot. 0x11 is used to avoid having bytes with
+      // zeroes in their hex representation.
+      const interaction1 = {
+        target: ethers.utils.getAddress(fillDistinctBytes(20, 0x11)),
+        callData: fillDistinctBytes(42, 0x11 + 20),
+      };
+      const interaction2 = {
+        target: ethers.utils.getAddress(fillDistinctBytes(20, 0x11 + 20 + 42)),
+        callData: fillDistinctBytes(1337, 0x11 + 2 * 20 + 42),
+      };
+      const interaction3 = {
+        target: ethers.utils.getAddress(
+          fillDistinctBytes(20, 0x11 + 2 * 20 + 42 + 1337),
+        ),
+        // Note: if the interaction decoding becomes significantly more
+        // inefficient, this test might fail with "tx has a higher gas limit
+        // than the block". In this case, the number of bytes below should be
+        // reduced.
+        callData: fillDistinctBytes(12000, 0x11 + 3 * 20 + 42 + 1337),
+      };
+
+      const encoder = new SettlementEncoder(testDomain);
+      await encoder.encodeInteraction(interaction1);
+      await encoder.encodeInteraction(interaction2);
+      await encoder.encodeInteraction(interaction3);
+
+      const numInteractions = 3;
+      const [decodedInteractions] = await encoding.decodeInteractionsTest(
+        encoder.encodedInteractions,
+        numInteractions,
+      );
+
+      expect(decodedInteractions.length).to.equal(3);
+      expect(decodedInteractions[0].target).to.equal(interaction1.target);
+      expect(decodedInteractions[0].callData).to.equal(interaction1.callData);
+      expect(decodedInteractions[1].target).to.equal(interaction2.target);
+      expect(decodedInteractions[1].callData).to.equal(interaction2.callData);
+      expect(decodedInteractions[2].target).to.equal(interaction3.target);
+      expect(decodedInteractions[2].callData).to.equal(interaction3.callData);
+    });
+
+    it("encoder fails to add interactions with too much calldata", async () => {
```

Comment:
This test takes very long to run for some reason. Unsure if it should be removed from this PR.
```
✓ encoder fails to add interactions with too much calldata (5681ms)
```

Reply:
Can we `.skip` it?